packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -958,6 +968,9 @@ function updateFunctionComponent(
 
   let nextChildren;
   prepareToReadContext(workInProgress, renderLanes);
+  if (enableSchedulingProfiler) {
+    markComponentRenderStarted(workInProgress);
```

Comment:
so, you basically already know all the places where "render starts" and "render stops", and we're adding them all over, right?

Reply:
Yes.